core/src/main/java/org/elasticsearch/search/aggregations/bucket/range/ipv4/IPv4RangeBuilder.java

```diff
@@ -148,7 +152,7 @@ public IPv4RangeBuilder addUnboundedFrom(String from) {
         int to = from + (~mask);
         long longTo = intIpToLongIp(to) + 1; // we have to +1 here as the range is non-inclusive on the "to" side
         if (longTo == InternalIPv4Range.MAX_IP) {
-            longTo = -1;
+            longTo =  InternalIPv4Range.MAX_IP - 1;
```

Comment:
Setting the value to `-1` leads to `null` as the to IP address. I assumed the expected IP address should be 255.255.255.255. But now that I rethink it there is a potential issue with that. As it is mentioned on line 153 the range is non inclusive. So it should be 256.0.0.0 which also correlates with `InternalIPv4Range.MAX_IP`. But I think this is actually not a valid IP address which is the reason why the system converts it to `null`. Going with the above solution would exclude the IP address 255.255.255.255.

As it is mentioned in the docs block `-1` means unbound end so you are right and the above change should not be needed. I will think of a test to verify that if the IP is 255.255.255.255 it is inside the range and will update the pull request.

As `-1` and `InternalIPv4Range.MAX_IP` are both converted to `null` I will check if the if clause is even needed.